src/api/Metadata.js

```diff
@@ -1053,6 +1054,74 @@ class Metadata extends File {
 
         return getProp(suggestionsResponse, 'data.suggestions', []);
     }
+
+    getMetadataOptionsUrl(scope: string, templateKey: string, fieldKey: string): string {
+        return `${this.getBaseApiUrl()}/metadata_templates/${scope}/${templateKey}/fields/${fieldKey}/options`;
+    }
+
+    /**
+     * Gets options associated with a taxonomy field.
+     *
+     * @param id
+     * @param scope
+     * @param templateKey
+     * @param fieldKey
+     * @param level
+     * @param options
+     * @returns {Promise<MetadataOptions>}
+     */
+    async getMetadataOptions(
+        id: string,
+        scope: string,
+        templateKey: string,
+        fieldKey: string,
+        level: number,
+        options: { marker?: string, searchInput?: string, signal?: AbortSignal },
+    ) {
+        this.errorCode = ERROR_CODE_FETCH_METADATA_OPTIONS;
+
+        if (!id) {
+            throw getBadItemError();
+        }
+
+        if (!scope) {
+            throw new Error('Missing scope');
+        }
+
+        if (!templateKey) {
+            throw new Error('Missing templateKey');
+        }
+
+        if (!fieldKey) {
+            throw new Error('Missing fieldKey');
+        }
+
+        // 0 is a valid level value
+        if (!level && level !== 0) {
+            throw new Error('Missing level');
+        }
+
+        const url = this.getMetadataOptionsUrl(scope, templateKey, fieldKey);
+        const { marker, searchInput, signal } = options;
+        const params = {
+            ...(marker ? { marker } : {}),
+            ...(searchInput ? { searchInput } : {}),
+        };
+
+        if (signal?.aborted) {
+            this.xhr.abort();
+
+            return {};
```

Comment:
Not completely sure what return value we'd want when aborted. Ill know more after integration

Reply:
This follows the async/await consumer pattern described here https://github.com/box/box-ui-elements/pull/3571/files#r1674082121. Without an early return, my unit tests start failing as the request is canceled but the `.get` method is called anyways.

Instead of returning, I think the proper thing is to throw the expected DOM exception typically emitted during this event